lvsfunc/comparison.py

```diff
@@ -2,17 +2,347 @@
+    :param arrangement:     2-dimension array (list of lists) of 0s and 1s representing a list of rows of clips(1)
+                            or blank spaces(0) (Default: None)
+    :param label_alignment: An integer from 1-9, corresponding to the positions of the keys on a numpad.
+                            Only used if `clips` is a dict.
+                            Determines where to place clip name using ``vapoursynth.VideoNode.text.Text`` (Default: 7)
+    """
+
+    def __init__(self,
+                 clips: Union[Dict[str, vs.VideoNode], Sequence[vs.VideoNode]],
+                 /,
+                 *,
+                 arrangement: Optional[List[List[int]]] = None,
+                 label_alignment: int = 7
+                 ) -> None:
+        super().__init__(clips, label_alignment=label_alignment)
+
+        if not self.width or not self.height:
+            raise ValueError("Tile: all clip widths, and heights must be the same")
+
+        if arrangement is None:
+            self.arrangement = self._auto_arrangement()
+        else:
+            is_one_dim = len(arrangement) < 2 or all(len(row) == 1 for row in arrangement)
+            if is_one_dim:
+                raise ValueError("Tile: use Stack instead if the array is one dimensional")
+            self.arrangement = arrangement
+
+        self.blank_clip = core.std.BlankClip(clip=self.clips[0], keep=1)
+
+        if not all(len(row) == (max_length := max(len(row) for row in self.arrangement)) for row in self.arrangement):
+            for row in self.arrangement:
+                if len(row) != max_length:
+                    diff = max_length - len(row)
+                    for _ in range(diff):
+                        row.append(0)  # padding all rows to the same length
+
+        array_count = 0
+        for list_ in self.arrangement:
+            for elem in list_:
+                if elem:
+                    array_count += 1
+
+        if array_count != self.num_clips:
+            raise ValueError('specified arrangement has an invalid number of clips')
+
+    def _compare(self) -> vs.VideoNode:
+        clips = self._marked_clips()
+        rows = [core.std.StackHorizontal([clips.pop(0) if elem else self.blank_clip for elem in row])
+                for row in self.arrangement]
+        return core.std.StackVertical(rows)
+
+    def _auto_arrangement(self) -> List[List[int]]:
+        dimension = 1 + math.isqrt(self.num_clips - 1)
+        row = [1 for _ in range(dimension)]
+        array = [row.copy() for _ in range(dimension)]
+
+        num_blank_clips = dimension ** 2 - self.num_clips
+
+        if num_blank_clips >= dimension:
+            array.pop(-1)
+            for i in range(1, num_blank_clips - dimension + 1):
+                array[-1][-i] = 0
+        else:
+            for i in range(1, num_blank_clips + 1):
+                array[-1][-i] = 0
+
+        return array
```

Comment:
```suggestion
    def _auto_arrangement(self) -> List[List[int]]:
        dimension = 1 + math.isqrt(self.num_clips - 1)
        cells = [1] * self.num_clips + [0] * (dimension ** 2 - self.num_clips)
        return list(map(list, grouper(array, dimension)))
```

grouper is an itertools recipe. Can be either copied to the file or imported from `more-itertools`.